components/ai_chat/core/browser/engine/engine_consumer_conversation_api.cc

```diff
@@ -124,7 +124,12 @@ void EngineConsumerConversationAPI::GenerateAssistantResponse(
     }
     ConversationEvent event;
     event.role = message->character_type;
-    event.content = (message == last_turn) ? human_input : message->text;
+
+    const std::string& text = (message->edits && !message->edits->empty())
```

Comment:
Optional:

I wonder if instead of having `edits` and `text` `text` should just return whatever is on the top of the edit stack?

Then we wouldn't need to do `edits.empty() ? text : edits.back()` every time

Optional 'cause it'd be quite a change in this PR and is a bit unrelated :smile: